### PR TITLE
Add: Assign thingId transform for use by Special Reports

### PR DESCRIPTION
**Pull request by andygout**
cc @ironsidevsquincy  @bjfletcher

### CODE REVIEW

File: server/config/profiles.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const podcasts = require('n-podcast-mapping').primaryTags();
 
 const articleModel = require('../transforms/article-model');
+const assignThingId = require('../transforms/assign-thing-id');
```
Comment (ironsidevsquincy) [inline server/config/profiles.js]: is this the best name for this transform? how about what it's doing, e.g. `addXMostRead` (whatever most read it is)

File: server/transforms/assign-thing-id.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = () => {
+	return class {
+		pre (req) {
+			if (req.url === '/special-reports') req.params.thingId = 'MTA=-R2VucmVz';
```
Comment (ironsidevsquincy) [inline server/transforms/assign-thing-id.js]: won't the url always have that, as the profile above is already doing this test

File: server/config/profiles.js (hunk 1 above)
Comment (ironsidevsquincy) [inline server/config/profiles.js]: Or maybe just `assign-X-to-thing-id`

What's is the `thingId` anyway?

File: server/transforms/assign-thing-id.js (hunk 2 above)
Comment (andygout) [inline server/transforms/assign-thing-id.js]: True, and I suppose in the instance of another profile using this transform we'd only then need to worry about customising for respective profiles.

File: server/config/profiles.js (hunk 1 above)
Comment (andygout) [inline server/config/profiles.js]: Breaking it down, it's `genreId` used by its non-vanity URL (`/stream/genreId/MTA=-R2VucmVz`), which (in this context) is assigned to the conceptId for use in the GraphQL query to acquire its Most Read articles.

On that basis do we want something like `assign-graphql-id-value-to-thing-id` / `assign-genre-id-value-to-thing-id` / `assign-url-value-to-thing-id`?

**Pull request merged**